### PR TITLE
chore: Merge Build,Lint and Unit Tests workflows

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Per our internal discussion to reduce the number of status checks per PR

### CODE REVIEW

File: .github/workflows/test_unit.yml [hunk 1]
```diff
@@ -19,13 +19,28 @@ jobs:
...
-  test_unit:
-    name: Unit Tests
+  test:
+    name: Test
     needs: resolve-modules
     strategy:
       matrix: ${{ fromJson(needs.resolve-modules.outputs.matrix) }}
       fail-fast: false
     runs-on: ubuntu-latest
+    services:
```
Comment (erezrokah) [inline .github/workflows/test_unit.yml:29]: This was removed in https://github.com/cloudquery/cloudquery/pull/2074.

I suggest we keep it here at least until we ensure all CI passes without it

**Review by hermanschaaf — APPROVED** [review]

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Let's hold off with that a bit. I would like to try a backward solution that I'll share today/tomorrow where there is one check that is the required check and it waits for the checks to pass based on the same logic that we already have via changed files.

**Comment by erezrokah** [comment]
@yevgenypats maybe merge this so at least we have a passing CI :) We can always do another PR with your change

**Comment by erezrokah** [comment]
Closing for https://github.com/cloudquery/cloudquery/pull/2093

**Pull request closed without merging**